src/ctap/crypto_wrapper.rs

```diff
@@ -204,17 +278,22 @@ pub fn encrypt_key_handle(
     let master_keys = storage::master_keys(env)?;
     let aes_enc_key = crypto::aes256::EncryptionKey::new(&master_keys.encryption);
 
-    let mut encrypted_id = match private_key {
+    let mut plaintext = [0; 64];
+    let version;
+    match private_key {
         PrivateKey::Ecdsa(ecdsa_key) => {
-            let mut plaintext = [0; 64];
             ecdsa_key.to_bytes(array_mut_ref!(plaintext, 0, 32));
-            plaintext[32..64].copy_from_slice(application);
-            let mut encrypted_id = aes256_cbc_encrypt(env.rng(), &aes_enc_key, &plaintext, true)?;
-            // Version number
-            encrypted_id.insert(0, 0x01);
-            encrypted_id
+            version = ECDSA_CREDENTIAL_ID_VERSION;
+        }
+        #[cfg(feature = "with_ed25519")]
+        PrivateKey::Ed25519(keypair) => {
+            plaintext[0..32].copy_from_slice(&keypair.secret.to_bytes());
```

Comment:
Documenting my thoughts here (not actionable): As long as all private keys are 32 bytes, one optoin is the `PrivateKey.to_bytes()` function and to check its output length to match the 32 byte (which should always succeed). But yours is more extensible, so I like it more.